posthog/api/dashboards/dashboard.py

```diff
@@ -434,6 +446,16 @@ def move_tile(self, request: Request, *args: Any, **kwargs: Any) -> Response:
         )
         return Response(serializer.data)
 
+    @action(methods=["POST"], detail=False)
+    def create_from_template_json(self, request: Request, *args: Any, **kwargs: Any) -> Response:
+        # load template from string into json
+        template = request.data["template"]
+
+        dashboard = Dashboard.objects.create(team_id=self.team_id)
```

Comment:
Since we have a schema.json we can use that to validate the received template here. So we can give users better feedback if they send invalid templates (accidentally or maliciously)

Reply:
What's the best practice for the json being shared across the frontend and backend?

Reply:
I think we can have the schema json in the backend behind an API endpoint and load that into the editor.

See https://github.com/microsoft/monaco-editor/issues/1331#issuecomment-470440651

So, we'd have something like

```
monaco.languages.json.jsonDefaults.setDiagnosticsOptions({ validate: true, allowComments: false, schemas: [], enableSchemaRequest: true});

monaco.editor.create(document.getElementById("container"), {
	value: '{\n  "$schema": "https://app.posthog.com/api/blah/etc/dashboard_templates/schema.json",\n  "author": {}\n}',
	language: "json"
});
```

Since it's in the back-end it can be a string constant or similar to be referenced in validation.

Then have something (probably) in the Serializer that can use that to validate template requests...